### ISSUE TITLE
Make Type Provider troubleshooting easier with better error reporting

### ISSUE THREAD

**Issue by smoothdeveloper** [Microsoft/visualfsharp#4956]
Can we consider replacing `e.Message` with `string e` at this location where we report exceptions happening at design time.

https://github.com/Microsoft/visualfsharp/blob/cea9284349e6ecc803cc39b67096b83010f0416d/src/fsharp/tainted.fs#L103

The exception details would be valuable in troubleshooting issues in type providers.

**Comment by dsyme** [comment]
@smoothdeveloper Yes, we can.  Please submit a PR for it - some test cases will need updating, that's all.

**Comment by smoothdeveloper** [comment]
Ok thanks.

The error will look scary, but it will definitely help troubleshooting.

![image](https://user-images.githubusercontent.com/87944/40404403-7810508c-5e0c-11e8-9e79-a5919ddc97ea.png)

**Comment by dsyme** [comment]
@smoothdeveloper We'll need to think about this more - exceptions are how TPs currently raise diagnostics, and we can't attach stack traces to those diagnostics.

**Comment by smoothdeveloper** [comment]
@dsyme, sorry I somehow missed your comment.

Can you expand on the meaning of "we can't attach stack traces to those diagnostics"? Is it related to how VS tooling display the errors or some policy about errors?

The PR is calling `ToString` on the exception so we are technically not attaching a stack trace object and just providing a string, but you probably mean it shouldn't surface even in textual form.

**Comment by smoothdeveloper** [comment]
another issue: https://github.com/rspeele/Rezoom.SQL/issues/54
